include/swift/Runtime/Config.h

```diff
@@ -38,7 +38,7 @@
 #ifndef SWIFT_USE_SWIFTCALL
 // Clang doesn't support mangling functions with the swiftcall attribute
 // on Windows and crashes during compilation: http://bugs.llvm.org/show_bug.cgi?id=32000
-#if (__has_attribute(swiftcall) || defined(__linux__)) && !defined(_WIN32)
+#if __has_attribute(swiftcall) && !defined(_WIN32)
```

Comment:
I .. don't entirely get what @jckarter and @rjmccall are saying; the change as written _does_ remove the runtime-header-included-in-irgen (in favour of ABI header), and as far as I can tell, the runtime _is_ built with the just-built clang (the issue was that the swift compiler is not -- it's built with system clang -- and there can be some divergence between the two).

I suspect I'm missing some subtlety here; maybe just that the non-swiftcc case never works, so we shouldn't support it at all?